addons/portal/static/src/js/portal_composer.js

```diff
@@ -138,15 +137,38 @@ var PortalComposer = publicWidget.Widget.extend({
         });
     },
     /**
-     * Returns a Promise that is never resolved to prevent sending the form
-     * twice when clicking twice on the button, in combination with the `async`
-     * in the event definition.
+     * prepares data to send message
      *
      * @private
+     */
+    _prepareMessageData: function () {
+        return _.extend(this.options || {}, {
+            'message': this.$('textarea[name="message"]').val(),
+            'attachment_ids': _.pluck(this.attachments, 'id'),
+            'attachment_tokens': _.pluck(this.attachments, 'access_token'),
+        });
+    },
+    /**
+     * Send message using rpc call and display new message and message count
+     *
+     * @private
+     * @param {Event} ev
      * @returns {Promise}
      */
-    _onSubmitButtonClick: function () {
-        return new Promise(function (resolve, reject) {});
+    _onSubmitButtonClick: async function (ev) {
+        ev.preventDefault();
+        if (!this.$inputTextarea.val().trim() && !this.attachments.length) {
+            this.$inputTextarea.addClass('border-danger');
+            let error = _t('Oops! Some fields are required. Please make sure that message or attachment is not empty!');
+            this.$(".o_portal_chatter_composer_error").text(error).removeClass('d-none');
+            return Promise.reject();
```

Comment:
Nothing can handle this promise's rejection, because it's an event handler. This is equivalent to `throw undefined`. Just use a simple `return;` here.